android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginViewModel.kt

```diff
@@ -77,17 +73,33 @@ constructor(
   val responseBodyHandler =
     object : ResponseHandler<ResponseBody> {
       override fun handleResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
-        if (response.isSuccessful) {
-          response.body()?.let {
-            with(it.string().decodeJson<UserInfo>()) {
-              sharedPreferences.write(USER_INFO_SHARED_PREFERENCE_KEY, this.encodeJson())
-              fetchLoggedInPractitioner(this)
-            }
-          }
-        } else {
```

Comment:
Does this mean we always fetch practitioner details regardless of whether the login was successful or not?

Reply:
Now that this check is deleted?

Reply:
No actually.

If you check the added changes below.

You will see, if the response is not successful, then handle errors and then `return`

Which means it will not fetch the practitioner details.

Reply:
Okay